cli/command/container/hijack.go

```diff
@@ -8,14 +8,19 @@ import (
 	"github.com/Sirupsen/logrus"
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/ioutils"
 	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/docker/docker/pkg/term"
 	"golang.org/x/net/context"
 )
 
+// The default escape key sequence: ctrl-p, ctrl-q
+var defaultEscapeKeys = []byte{16, 17}
+
 // holdHijackedConnection handles copying input to and output from streams to the
 // connection
 // nolint: gocyclo
-func holdHijackedConnection(ctx context.Context, streams command.Streams, tty bool, inputStream io.ReadCloser, outputStream, errorStream io.Writer, resp types.HijackedResponse) error {
+func holdHijackedConnection(ctx context.Context, streams command.Streams, tty bool, detachKeys string, inputStream io.ReadCloser, outputStream, errorStream io.Writer, resp types.HijackedResponse) error {
```

Comment:
It seems like there are cases where in/out/err are set to nil based on flags, but I think we could still use `streams`. We just need a struct which implements that interface and we can use it to build up the correct set of streams for those cases.